src/policy/mod.rs

```diff
@@ -37,6 +37,8 @@ pub use self::concrete::Policy as Concrete;
 pub use self::semantic::Policy as Semantic;
 use MiniscriptKey;
 
+/// Policy entailment algorithm maximum number of terminals allowed
+const ENTAILMENT_MAX_TERMINALS: usize = 24;
```

Comment:
Some quick benchmarks on map laptop show that this allows an entailment to take nearly 10 seconds in debug mode, 0.3s in release mode. I'd prefer this value be reduced to 20 so that everything stays under a second.